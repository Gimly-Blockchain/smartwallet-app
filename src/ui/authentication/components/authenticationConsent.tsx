import React from 'react'
import { ButtonSection } from 'src/ui/structure/buttonSectionBottom'
import { Text, StyleSheet, View } from 'react-native'
import I18n from 'src/locales/i18n'
import strings from '../../../locales/strings'
import { Colors, Typography, Spacing } from 'src/styles'
import { IssuerCard } from '../../documents/components/issuerCard'
import { AuthenticationRequestSummary } from '../../../actions/sso/types'

interface Props {
  authenticationDetails: AuthenticationRequestSummary
  generateAndSendAuthenticationResponse: Function
  cancelAuthenticationRequest: Function
}

interface State {}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.backgroundLightMain,
  },
  topSection: {
    flex: 0.9,
  },
  issuerCard: {
    marginTop: Spacing.LG,
  },
  authRequestContainer: {
    flex: 1,
    paddingHorizontal: '10%',
    justifyContent: 'flex-start',
    alignItems: 'center',
    marginTop: Spacing.XL,
  },
  authRequestText: {
    ...Typography.baseFontStyles,
    ...Typography.centeredText,
    fontSize: Typography.textLG,
    marginTop: Spacing.SM,
  },
  buttonSection: {
    flex: 0.1,
  },
})

export class AuthenticationConsentComponent extends React.Component<
  Props,
  State
> {
  public state = {
    pending: false,
  }

  private handleConfirm = () => {
    this.setState({ pending: true })
    return this.props.generateAndSendAuthenticationResponse()
  }

  public render() {
    const { requester, description } = this.props.authenticationDetails
    return (
      <View style={styles.container}>
        <View style={styles.topSection}>
          <IssuerCard issuer={requester} style={styles.issuerCard} />
          <View style={styles.authRequestContainer}>
            <Text style={styles.authRequestText}>
              {I18n.t(strings.WOULD_YOU_LIKE_TO)}
            </Text>
            <Text
              style={[styles.authRequestText, { fontSize: Typography.text4XL }]}
            >
              {description}
            </Text>
            <Text style={styles.authRequestText}>
              {I18n.t(strings.WITH_YOUR_SMARTWALLET)}
            </Text>
          </View>
        </View>
        <View style={styles.buttonSection}>
          <ButtonSection
            disabled={this.state.pending}
            denyDisabled={this.state.pending}
            confirmText={I18n.t(strings.AUTHORIZE)}
            denyText={I18n.t(strings.DENY)}
            handleConfirm={this.handleConfirm}
            handleDeny={() => this.props.cancelAuthenticationRequest()}
          />
        </View>
      </View>
    )
  }
}
