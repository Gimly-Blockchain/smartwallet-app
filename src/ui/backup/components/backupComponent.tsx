import * as React from 'react'
import { ScrollView, StyleSheet, Text, View } from 'react-native'
import { Container } from '../../structure'
import { Colors } from '../../../styles'
import { fontMain } from '../../../styles/typography'
import { GradientButton } from '../../structure/gradientButton'
import { NavigationSection } from '../../structure/navigationSection'

interface Props {
  isLoading: boolean
  isAutoBackupEnabled: boolean
  enableAutoBackup: () => void
  onDisableAutoBackup: () => void
  exportBackup: () => void
  goBack: () => void
  lastBackup?: string
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.baseBlack,
    justifyContent: 'flex-start',
    alignItems: 'stretch',
  },
  scrollView: {
    padding: 18,
  },
  section: {
    backgroundColor: Colors.black,
    justifyContent: 'flex-start',
    alignItems: 'stretch',
    marginTop: 32,
    borderRadius: 8,
    padding: 24,
    paddingBottom: 28,
  },
  title: {
    fontFamily: fontMain,
    color: Colors.white,
    fontSize: 28,
  },
  subtitle: {
    fontFamily: fontMain,
    color: Colors.white,
    fontSize: 22,
    textAlign: 'center',
  },
  description: {
    color: Colors.white050,
    fontSize: 16,
    marginTop: 8,
    textAlign: 'center',
  },
  warning: {
    color: Colors.error,
    fontSize: 14,
    textAlign: 'center',
    marginTop: 28,
  },
  info: {
    textAlign: 'center',
    color: Colors.purpleMain,
    marginTop: 28,
  },
  buttonContainer: {
    marginTop: 22,
    width: '100%',
    paddingVertical: 9,
    margin: 0,
    borderRadius: 8,
  },
  buttonText: {
    fontSize: 18,
  },
})
const BackupComponent: React.FC<Props> = ({
  isLoading,
  enableAutoBackup,
  onDisableAutoBackup,
  exportBackup,
  isAutoBackupEnabled,
  goBack,
  lastBackup,
}) => {
  return (
    <Container style={styles.container}>
      <NavigationSection
        onNavigation={goBack}
        isBackButton={true}
        isDark={true}
      />
      <ScrollView style={styles.scrollView}>
        <Text style={styles.title}>Backup options</Text>

        <View style={styles.section}>
          <Text style={styles.subtitle}>Backup data on your own</Text>
          <Text style={styles.description}>
            Download an encrypted copy of the data in your SmartWallet to your
            device. Make sure to keep your backups up-to-date and stored
            somewhere safe
          </Text>
          <GradientButton
            containerStyle={styles.buttonContainer}
            textStyle={styles.buttonText}
            onPress={exportBackup}
            text={'Export file'}
            disabled={isLoading}
          />
          {lastBackup && (
            <Text style={styles.info}>
              {`Last backup ${new Date(lastBackup).toLocaleDateString()}`}
            </Text>
          )}
        </View>

        <View style={styles.section}>
          <Text style={styles.subtitle}>Jolocom backup service</Text>
          <Text style={styles.description}>
            Store your information using Jolocom’s backup service. With this
            option, the Jolocom team cannot view your data and your info stays
            automatically up-to-date.
          </Text>
          {isAutoBackupEnabled ? (
            <React.Fragment>
              <GradientButton
                containerStyle={styles.buttonContainer}
                textStyle={styles.buttonText}
                text={isLoading ? 'Deleting backup...' : 'Disable auto-backup'}
                onPress={onDisableAutoBackup}
                disabled={isLoading}
              />
              {lastBackup && (
                <Text style={styles.info}>
                  {`Last backup ${new Date(lastBackup).toLocaleDateString()}`}
                </Text>
              )}
            </React.Fragment>
          ) : (
            <React.Fragment>
              <GradientButton
                containerStyle={styles.buttonContainer}
                textStyle={styles.buttonText}
                onPress={enableAutoBackup}
                text={
                  isLoading ? 'Synchronization started…' : 'Use backup service'
                }
                disabled={isLoading}
              />
              <Text style={styles.warning}>
                By clicking here i agreed backing up my data on Jolocom’s
                central server and i know that it is not viewable by anyone
              </Text>
            </React.Fragment>
          )}
        </View>
      </ScrollView>
    </Container>
  )
}

export default BackupComponent
