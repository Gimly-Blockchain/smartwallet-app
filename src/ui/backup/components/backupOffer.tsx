import * as React from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import { Container } from '../../structure'
import { Colors } from '../../../styles'
import { GradientButton } from '../../structure/gradientButton'
import { ActionSheet } from '../../structure/actionSheet'

interface Props {
  enableAutoBackup: () => void
  manualBackup: () => void
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.baseBlack,
    justifyContent: 'flex-start',
    alignItems: 'stretch',
  },
  title: {
    color: Colors.white080,
    fontSize: 22,
    textAlign: 'center',
    marginHorizontal: 40,
  },
  logoTitle: {
    color: Colors.white080,
    fontSize: 22,
    textAlign: 'center',
    marginHorizontal: 108,
  },
  bottomSheet: {
    position: 'absolute',
    bottom: 0,
    backgroundColor: Colors.black,
    height: 230,
    width: '100%',
    borderRadius: 22,
    borderBottomLeftRadius: 0,
    borderBottomRightRadius: 0,
    paddingTop: 28,
    paddingHorizontal: 20,
  },
  button: {
    marginTop: 28,
    marginHorizontal: 20,
  },
  info: {
    color: Colors.white050,
    fontSize: 14,
    textAlign: 'center',
    marginTop: 10,
  },
  textButton: {
    color: Colors.white,
    fontSize: 20,
    textAlign: 'center',
    marginTop: 20,
  },
})
export const BackupOfferComponent: React.FC<Props> = ({
  enableAutoBackup,
  manualBackup,
}) => {
  return (
    <Container style={styles.container}>
      <Text style={styles.title}>
        Store your data by yourself or try to use Jolocom backup service!
      </Text>
      <View>{/* TODO add correct icon*/}</View>
      <Text style={styles.logoTitle}>Automatic synchronization</Text>
      <ActionSheet showSlide={true}>
        <GradientButton
          onPress={enableAutoBackup}
          text={'Turn on backup service'}
        />
        <Text style={styles.info}>
          By clicking here i agreed backing up my data on Jolocomâ€™s central
          server server and i know that it is not viewable by anyone
        </Text>
        <TouchableOpacity onPress={manualBackup}>
          <Text style={styles.textButton}>Keep it manually</Text>
        </TouchableOpacity>
      </ActionSheet>
    </Container>
  )
}
