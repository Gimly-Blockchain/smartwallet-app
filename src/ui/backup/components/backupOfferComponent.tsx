import * as React from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import { Container } from '../../structure'
import { Colors } from '../../../styles'
import { NavigationSection } from '../../structure/navigationSection'
import { GradientButton } from '../../structure/gradientButton'

interface Props {
  enableAutoBackup: () => void
  close: () => void
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.baseBlack,
    justifyContent: 'flex-start',
    alignItems: 'stretch',
  },
  title: {
    color: Colors.white080,
    fontSize: 22,
    textAlign: 'center',
    marginHorizontal: 40,
  },
  logoTitle: {
    color: Colors.white080,
    fontSize: 22,
    textAlign: 'center',
    marginHorizontal: 108,
  },
  bottomSheet: {
    position: 'absolute',
    bottom: 0,
    backgroundColor: Colors.black,
    height: 230,
    width: '100%',
    borderRadius: 22,
    borderBottomLeftRadius: 0,
    borderBottomRightRadius: 0,
    paddingTop: 28,
    paddingHorizontal: 20,
  },
  button: {
    marginTop: 28,
    marginHorizontal: 20,
  },
  info: {
    color: Colors.white050,
    fontSize: 14,
    textAlign: 'center',
    marginTop: 10,
  },
  textButton: {
    color: Colors.white,
    fontSize: 20,
    textAlign: 'center',
    marginTop: 20,
  },
})
const BackupOfferComponent: React.FC<Props> = ({ close, enableAutoBackup }) => {
  return (
    <Container style={styles.container}>
      <NavigationSection
        onNavigation={close}
        isBackButton={false}
        isDark={true}
      />
      <Text style={styles.title}>
        Store your data by yourself or try to use Jolocom backup service!
      </Text>
      <View>{/* TODO add correct icon*/}</View>
      <Text style={styles.logoTitle}>Automatic synchronization</Text>
      <View style={styles.bottomSheet}>
        <GradientButton
          onPress={enableAutoBackup}
          text={'Turn on backup service'}
        />
        <Text style={styles.info}>
          By clicking here i agreed backing up my data on Jolocomâ€™s central
          server and i know that it is not viewable by anyone
        </Text>
        {/*TODO add onPress*/}
        <TouchableOpacity>
          <Text style={styles.textButton}>Keep it manually</Text>
        </TouchableOpacity>
      </View>
    </Container>
  )
}

export default BackupOfferComponent
