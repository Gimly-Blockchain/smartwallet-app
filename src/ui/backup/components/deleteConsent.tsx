import { StyleSheet, Text, View } from 'react-native'
import { Colors } from '../../../styles'
import React from 'react'

const loaders = require('react-native-indicator')

interface Props {
  isLoading: boolean
  onCancel: () => void
  onAgree: () => void
}
const styles = StyleSheet.create({
  title: {
    color: Colors.white,
    fontSize: 26,
  },
  info: {
    color: Colors.white,
    fontSize: 14,
    marginTop: 8,
  },
  loaderContainer: {
    alignItems: 'center',
  },
  buttonSection: {
    marginTop: 20,
    flexDirection: 'row',
    justifyContent: 'flex-end',
  },
  button: {
    margin: 10,
    color: Colors.white,
    fontSize: 18,
  },
})
export const DeleteConsent: React.FC<Props> = ({
  isLoading,
  onCancel,
  onAgree,
}) => (
  <React.Fragment>
    <Text style={styles.title}>{'Disable auto-backup'}</Text>
    {isLoading ? (
      <View style={styles.loaderContainer}>
        <loaders.RippleLoader size={42} strokeWidth={2} color={Colors.beige} />
      </View>
    ) : (
      <React.Fragment>
        <Text style={styles.info}>
          {
            'By disabling this function you will erase all information from our server'
          }
        </Text>
        <View style={styles.buttonSection}>
          <Text style={styles.button} onPress={onCancel}>
            {'Cancel'}
          </Text>
          <Text style={styles.button} onPress={onAgree}>
            {'Agree'}
          </Text>
        </View>
      </React.Fragment>
    )}
  </React.Fragment>
)
