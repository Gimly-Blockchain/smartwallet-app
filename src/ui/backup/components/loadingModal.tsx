import React from 'react'
import { StyleSheet, View } from 'react-native'
import { Colors } from '../../../styles'
import Modal from 'react-native-modal'

interface Props {
  isVisible: boolean
  close?: () => void
}

const styles = StyleSheet.create({
  container: {
    borderRadius: 8,
    backgroundColor: '#444145',
    justifyContent: 'center',
    alignItems: 'stretch',
    padding: 24,
  },
  title: {
    color: Colors.white,
    fontSize: 26,
    textAlign: 'center',
  },
  loaderContainer: {
    margin: 24,
  },
  waring: {
    color: Colors.error,
    fontSize: 14,
    textAlign: 'center',
  },
})
export const LoadingModal: React.FC<Props> = ({
  isVisible,
  close,
  children,
}) => (
  <Modal
    isVisible={isVisible}
    animationIn={'fadeIn'}
    animationOut={'fadeOut'}
    onBackdropPress={close}
  >
    <View style={styles.container}>{children}</View>
  </Modal>
)
