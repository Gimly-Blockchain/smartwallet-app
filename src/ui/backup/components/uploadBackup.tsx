import { StyleSheet, Text, View } from 'react-native'
import { Colors } from '../../../styles'
import React from 'react'
const loaders = require('react-native-indicator')

const styles = StyleSheet.create({
  title: {
    color: Colors.white,
    fontSize: 26,
    textAlign: 'center',
  },
  loaderContainer: {
    alignItems: 'center',
    margin: 24,
  },
  waring: {
    color: Colors.error,
    fontSize: 14,
    textAlign: 'center',
  },
})
export const UploadBackup = () => (
  <React.Fragment>
    <Text style={styles.title}>{'Backing-up with Jolocom'}</Text>
    <View style={styles.loaderContainer}>
      <loaders.RippleLoader size={42} strokeWidth={2} color={Colors.beige} />
    </View>
    <Text style={styles.waring}>
      {
        'The process of secure synchronization of current data and configuration of auto-update function has started'
      }
    </Text>
  </React.Fragment>
)
