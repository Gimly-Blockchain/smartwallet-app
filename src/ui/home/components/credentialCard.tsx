import React from 'react'
import {
  StyleSheet,
  View,
  TextStyle,
  GestureResponderEvent,
  ViewStyle,
} from 'react-native'
import { JolocomTheme } from 'src/styles/jolocom-theme'
import {
  ClaimCard,
  PlaceholderClaimCard,
  CollapsedCredentialCard,
} from 'src/ui/sso/components/claimCard'
import { ReactNode } from 'react-redux'
import { DecoratedClaims } from 'src/reducers/account'
import I18n from 'src/locales/i18n'
import { values, all, isEmpty } from 'ramda'

interface Props {
  handleInteraction?: (event: GestureResponderEvent) => void
  credentialItem: DecoratedClaims
  collapsible?: boolean
  leftIcon: ReactNode
  rightIcon?: ReactNode
  containerStyle?: ViewStyle
  claimCardStyle?: {
    primaryText?: TextStyle
    secondaryText?: TextStyle
  }
}

interface State {
  collapsed: boolean
}

export class CredentialCard extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)

    this.state = {
      collapsed: props.collapsible || false,
    }
  }

  private getStyles = () =>
    StyleSheet.create({
      defaultContainerStyle: {
        flexDirection: 'row',
        backgroundColor: JolocomTheme.primaryColorWhite,
        paddingVertical: 12,
        borderBottomWidth: 1,
        borderColor: 'rgb(236, 236, 236)',
      },
    })

  private toggleCollapse = () => {
    if (this.props.collapsible) {
      this.setState({ collapsed: !this.state.collapsed })
    }
  }

  private renderClaim = (credentialItem: DecoratedClaims) => {
    const { handleInteraction } = this.props
    const { credentialType, claimData } = credentialItem
    const onEdit = handleInteraction || (() => {})

    const isBlank = all(isEmpty, values(claimData))
    if (isBlank) {
      return (
        <PlaceholderClaimCard onEdit={onEdit} credentialType={credentialType} />
      )
    }
    return Object.keys(claimData).map(key => (
      <ClaimCard
        key={key}
        primaryText={I18n.t(claimData[key])}
        secondaryText={I18n.t(key)}
      />
    ))
  }

  private renderCollapsedClaim = (credentialItem: DecoratedClaims) => (
    <CollapsedCredentialCard
      title={I18n.t(credentialItem.credentialType)}
      values={Object.keys(credentialItem.claimData).map(
        k => credentialItem.claimData[k],
      )}
    />
  )

  render() {
    const { credentialItem, containerStyle, leftIcon, rightIcon } = this.props
    const { collapsed } = this.state
    const { defaultContainerStyle } = this.getStyles()

    return (
      <View
        style={[
          StyleSheet.flatten(defaultContainerStyle),
          containerStyle || {},
        ]}
      >
        <View flex={0.2} alignItems={'center'}>
          {leftIcon}
        </View>
        <View onTouchEnd={this.toggleCollapse} flex={0.7} overflow={'scroll'}>
          {collapsed
            ? this.renderCollapsedClaim(credentialItem)
            : this.renderClaim(credentialItem)}
        </View>
        <View flex={0.1} onTouchEnd={this.props.handleInteraction}>
          {rightIcon || null}
        </View>
      </View>
    )
  }
}
