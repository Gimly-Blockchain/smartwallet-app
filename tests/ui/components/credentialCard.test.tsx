import React from 'react'
import { CredentialCard } from 'src/ui/home/components/credentialCard'
import { shallow } from 'enzyme'

describe('ClaimCard component', () => {
  it('matches the snapshot on render', () => {
    const props = {
      credential: {
        credentialType: 'Name',
        claimData: {
          familyName: 'Running',
        },
        id: 'claim:id:1234',
        issuer: {
          did: 'did:issuer:ex',
        },
        subject: 'did:subject:ex',
      },
      leftIcon: null,
      did: 'test0123',

      openClaimsDetails: () => null,
    }
    const rendered = shallow(<CredentialCard {...props} />)
    expect(rendered).toMatchSnapshot()
  })

  it('matches the snapshot of a two line claim', () => {
    const props = {
      credential: {
        credentialType: 'Name',
        claimData: {
          givenName: 'Test',
          familyName: 'Running',
        },
        id: 'claim:id:1234',
        issuer: {
          did: 'did:issuer:ex',
        },
        subject: 'did:subject: ex',
      },
      leftIcon: null,
      did: 'test0123',
      openClaimsDetails: () => null,
    }

    const rendered = shallow(<CredentialCard {...props} />)
    expect(rendered).toMatchSnapshot()
  })
})
