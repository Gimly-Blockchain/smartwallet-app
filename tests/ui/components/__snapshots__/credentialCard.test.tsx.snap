// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClaimCard component matches the snapshot of a two line claim 1`] = `
<CardWrapper
  style={
    Object {
      "borderBottomWidth": 1,
      "borderTopWidth": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(5, 5, 13, 0.4)",
          "fontFamily": "TT Commons",
          "fontSize": 16,
        }
      }
    >
      Given Name
    </Text>
    <Text
      style={
        Object {
          "color": "rgb(5, 5, 13)",
          "fontFamily": "TT Commons",
          "fontSize": 22,
          "marginBottom": 4,
        }
      }
    >
      Test
    </Text>
    <Text
      style={
        Object {
          "color": "rgba(5, 5, 13, 0.4)",
          "fontFamily": "TT Commons",
          "fontSize": 16,
        }
      }
    >
      Family Name
    </Text>
    <Text
      style={
        Object {
          "color": "rgb(5, 5, 13)",
          "fontFamily": "TT Commons",
          "fontSize": 22,
          "marginBottom": 4,
        }
      }
    >
      Running
    </Text>
  </View>
</CardWrapper>
`;

exports[`ClaimCard component matches the snapshot on render 1`] = `
<CardWrapper
  style={
    Object {
      "borderBottomWidth": 1,
      "borderTopWidth": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginLeft": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(5, 5, 13, 0.4)",
          "fontFamily": "TT Commons",
          "fontSize": 16,
        }
      }
    >
      Family Name
    </Text>
    <Text
      style={
        Object {
          "color": "rgb(5, 5, 13)",
          "fontFamily": "TT Commons",
          "fontSize": 22,
          "marginBottom": 4,
        }
      }
    >
      Running
    </Text>
  </View>
</CardWrapper>
`;
